js/util/navigationUtil.ts

```diff
@@ -1,14 +1,48 @@
 export default class NavigationUtil {
-  static getActiveRouteName(navigationState: any): any {
-    if (!navigationState) {
+  // :TODO: (jmtaber129): Add typing for this ref.
+  private static heapNavRef: any;
+
+  static setNavigationRef(ref: any): void {
+    this.heapNavRef = ref;
+  }
+
+  static getScreenPropsForCurrentRoute(): {
+    path: string;
+    screenName: string;
+  } | null {
+    if (
+      !(this.heapNavRef && this.heapNavRef.state && this.heapNavRef.state.nav)
+    ) {
       return null;
     }
+
+    const routeProps = this.getActiveRouteProps(this.heapNavRef.state.nav);
+
+    if (routeProps) {
+      return routeProps;
+    }
+
+    return null;
+  }
+
+  // :TODO: (jmtaber129): Add type for navigationState.
+  static getActiveRouteProps(
```

Comment:
Perhaps return an array of screen strings, so that the last one is the current screen, and the rest can be glommed together to paths